packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -647,6 +661,24 @@ export class TrimergeClient<
   }
 
   public shutdown(): Promise<void> | void {
-    return this.store.shutdown();
+    if (this.isShutdown) {
+      throw new Error('already shutdown');
+    }
+    this.isShutdown = true;
+
+    const shutdownPromises: Promise<void>[] = [];
+
+    const storeShutdown = this.store.shutdown();
+    if (storeShutdown) {
```

Comment:
for style consistency with below:
```suggestion
    const storeShutdownPromise = this.store.shutdown();
    if (storeShutdownPromise) {
```